### PR TITLE
[SPARK-43081][FOLLOW-UP][ML][CONNECT] Make torch dataloader support torch 1.x

### PR DESCRIPTION
**Pull request by WeichenXu123**
### What changes were proposed in this pull request?

Make torch dataloader support torch 1.x.
Currently, when running with torch 1.x with num_workers > 0, an error is raised like:
```
ValueError: prefetch_factor option could only be specified in multiprocessing.let num_workers > 0 to enable multiprocessing.
```

### Why are the changes needed?

Compatibility fix.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manually run unit tests with torch 1.x

### CODE REVIEW

File: python/pyspark/ml/torch/distributor.py [hunk 1]
```diff
@@ -995,4 +995,9 @@ def _get_spark_partition_data_loader(
 
     dataset = _SparkPartitionTorchDataset(arrow_file, schema, num_samples)
 
-    return DataLoader(dataset, batch_size, num_workers=num_workers, prefetch_factor=prefetch_factor)
```
Comment (zhengruifeng) [inline python/pyspark/ml/torch/distributor.py:998]: mind adding a comment here?

**Review by zhengruifeng — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]
can you remove `.hypothesis/unicode_data/12.1.0/charmap.json.gz`? seems like a mistake.

otherwise lgtm

**Comment by WeichenXu123** [comment]
jenkins merge

**Comment by zhengruifeng** [comment]
merged into master

**Pull request closed without merging**